yarvt

```diff
@@ -594,7 +605,9 @@ env_check $@
 
 $@
 
-if [[ ${KEEP_LOGS} == 1 ]];then
+# If a command fails then logs are kept even if
+# ${KEEP_LOGS} == 0  in order to identify what went wrong
+if [[ ${KEEP_LOGS} == 1 || $? != 0 ]];then
```

Comment:
$? holds the return value of the last command, that also means the echo command from the debug/error message which will probably succeed, so you'll probably never keep logs.

Reply:
If `$? !=0 ` then the previous function/command failed and as a result logs are not deleted even if `${KEEP_LOGS} == 0`, please examine it more carefully it works fine on all cases.

Reply:
My point is that "something went wrong" doesn't always mean we'll get a return code other than 0, or that the last thing to be executed will be one of our functions. For now that's the case but in the future we may want to print warnings as well and/or do other things after running a function and before exiting / cleaning the logs (e.g. print a summary of how long it took for a command to be executed). It would be better if we just changed KEEP_LOGS = 1 inside pr_err and/or pr_wrn to achieve what you want.